### DISCUSSION TITLE
Why don’t the official submarines have wiring on the wrecks?

### DISCUSSION THREAD

**Discussion by KougamiYuriko** [FakeFishGames/Barotrauma#14856] [Ideas and suggestions]
I know very few people go to repair wrecks, but I enjoy repairing them. I can only do this with wrecks designed by other mod creators. The official ships don’t have wiring—why is that?
 I understand the official game didn’t consider the gameplay of docking and repairing wrecks, but modded ships have accomplished this. It’s such a fun project to completely connect to a wreck, and third-party mod wrecks have intact wiring; you just need to repair the electrical systems, drain the water, and lift the wreck back to the outpost. That’s also a joy!
I completely recommend that the officials redo the wrecks. The current wrecks are just empty shells with no exploration fun. Besides needing essential supplies, I have no reason to explore them. At the beginning of the game, when resources are scarce, there aren't any wrecks. By the time I reach the last map, I have too many resources, making it pointless to search the wrecks for supplies; it's just a waste of my time because I know the official wrecks are not recoverable.
Additionally, having too many wrecks generated on the last map causes a significant drop in frame rate. The wrecks contain a lot of items, and the frame rate drops from 60 to 15. If I use the editor to delete wrecks during a battle, the frame rate goes back up. The officials know that too many items can cause frame rate instability, yet they haven’t implemented any limits, which I don’t understand.

**Comment by SomeRandomNoobKekeke** [comment]
> Why don’t the official submarines have wiring on the wrecks?

For performance reasons, imagine all those wrecked machines actually trying to work

> having too many wrecks generated on the last map causes a significant drop in frame rat

That level will be removed in next update for that exact reason

> The officials know that too many items can cause frame rate instability

Items peacefully lying in cabinets don't cause lags, but working machines do

> yet they haven’t implemented any limits, which I don’t understand

There are some optimization for far from player stuff, but if you disable that completely pirates won't work

**Reply by KougamiYuriko** [reply]
> > 为什么官方潜艇不在沉船上布线？
> 
> 出于性能原因，想象一下所有这些失事的机器实际上都在努力工作
> 
> > 在最后一个地图上生成过多的 wreck 会导致 Frame RAT 显著下降
> 
> 正是出于这个原因，该级别将在下次更新中删除
> 
> > 官员们知道，项目太多会导致帧率不稳定
> 
> 安静地躺在橱柜里的物品不会导致滞后，但工作中的机器会导致滞后
> 
> > 但他们没有实施任何限制，我不明白
> 
> 对于远离玩家的东西有一些优化，但如果你完全禁用它，海盗将不起作用

Understood, thank you for your answer; you addressed my questions about the FPS and the sunken ship. But in the initial state of the sunken ship, there is no power, so the reactor and electrical systems won't work, so it shouldn't be a big problem, right?
